paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/NumberSequenceRowSource.java

```diff
@@ -0,0 +1,446 @@
+
+        private static final int CURRENT_VERSION = 1;
+
+        @Override
+        public int getVersion() {
+            return CURRENT_VERSION;
+        }
+
+        @Override
+        public byte[] serialize(Collection<NumberSequenceSplit> checkpoint) throws IOException {
+            // Each split needs 2 longs (16 bytes) plus the UFT representation of the string (2 +
+            // length)
+            // Assuming at most 4 digit split IDs, 22 bytes per split avoids any intermediate array
+            // resizing.
+            // plus four bytes for the length field
+            final DataOutputSerializer out = new DataOutputSerializer(checkpoint.size() * 22 + 4);
+            out.writeInt(checkpoint.size());
+            for (NumberSequenceSplit split : checkpoint) {
+                SplitSerializer.serializeV1(out, split);
+            }
+            return out.getCopyOfBuffer();
+        }
+
+        @Override
+        public Collection<NumberSequenceSplit> deserialize(int version, byte[] serialized)
+                throws IOException {
+            if (version != CURRENT_VERSION) {
+                throw new IOException("Unrecognized version: " + version);
+            }
+            final DataInputDeserializer in = new DataInputDeserializer(serialized);
+            final int num = in.readInt();
+            final ArrayList<NumberSequenceSplit> result = new ArrayList<>(num);
+            for (int remaining = num; remaining > 0; remaining--) {
+                result.add(SplitSerializer.deserializeV1(in));
+            }
+            return result;
+        }
+    }
+
+    private static class NumberSequenceIterator extends SplittableIterator<RowData> {
+
+        private static final long serialVersionUID = 1L;
+
+        /** The last number returned by the iterator. */
+        private final long to;
+
+        /** The next number to be returned. */
+        private long current;
+
+        /**
+         * Creates a new splittable iterator, returning the range [from, to]. Both boundaries of the
+         * interval are inclusive.
+         *
+         * @param from The first number returned by the iterator.
+         * @param to The last number returned by the iterator.
+         */
+        public NumberSequenceIterator(long from, long to) {
+            if (from > to) {
+                throw new IllegalArgumentException(
+                        "The 'to' value must not be smaller than the 'from' value.");
+            }
+
+            this.current = from;
+            this.to = to;
+        }
+
+        /**
+         * Internal constructor to allow for empty iterators.
+         *
+         * @param from The first number returned by the iterator.
+         * @param to The last number returned by the iterator.
+         * @param unused A dummy parameter to disambiguate the constructor.
+         */
+        private NumberSequenceIterator(long from, long to, boolean unused) {
```

Comment:
where the unused being used?

Reply:
`unused A dummy parameter to disambiguate the constructor`.

Just copy it from Flink for future using.